control/flags.go

```diff
@@ -19,19 +19,21 @@ limitations under the License.
 
 package control
 
-import "github.com/codegangsta/cli"
+import (
+	"fmt"
+
+	"github.com/codegangsta/cli"
+)
 
 var (
-	flNumberOfPLs = cli.IntFlag{
+	flNumberOfPLs = cli.StringFlag{
```

Comment:
yes, you are absolutely correct, @IRCody; the change to string flags is a change chosen to work around an obvious issue with the cli library we are using.

The reason I made the changes I made in this PR (swapping these fields over so that they are parsed as string flags and adding error handling to ensure they can still be used as integers or durations) is that I didn't feel that this change was nearly as drastic as a change to a different cli library would be (yes, there are others out there) and it resolves this issue without introducing the secondary issue I encountered and reported as issue #1016 (which is now closed, but which was triggered by the fact that there were default values set for these properties and that we couldn't tell the difference between a value that was set via an environment variable and a value that was set using the defined default value in this cli library; default values that were added in an earlier PR simply to avoid having bogus default values added to usage text that is output by that cli library in response to a `snapd --help` or `go run snapd.go --help` command).

This change to string flags seemed to be the smallest change that I could make and still resolve this issue without causing any side-effects that were unwanted. **If** we are going to switch over to another cli library, I think that should be handled in a separate PR that addresses the question of which cli library we should be using in Snap as a completely separate issue. Yes, the change to string flags for these fields is a work-around for an the fact that we can't get to the detail we need to resolve this issue through the codegangsta library itself and that library adds bogus default values to the help output if you don't define a default value for integer and duration values, but I don't see the harm in swapping over to parsing these as string flags, especially since I've added code to catch situations where the values passed using these flags cannot be parsed as the appropriate integer/duration type and report those values as invalid (logging a fatal error and exiting in that situation).

So, in short, I **don't** think we should switch to a different library at this time and I think the changes made in this PR **do** resolve the issue associated with this PR, all while making as few changes to the existing codebase as possible.